src/policy/packages.cpp

```diff
@@ -118,8 +118,9 @@ bool IsWellFormedPackage(const Package& txns, PackageValidationState& state, boo
 
 bool IsChildWithParents(const Package& package)
 {
+    if (package.empty()) return false;
     assert(std::all_of(package.cbegin(), package.cend(), [](const auto& tx){return tx != nullptr;}));
-    if (package.size() < 2) return false;
+    if (package.size() < 2) return true;
```

Comment:
Note: this also requires updating this documentation: https://github.com/bitcoin/bitcoin/blob/538ccaed004ff89ec8b2b71d8711feed624ffccc/doc/policy/packages.md?plain=1#L78-L79

Reply:
Worth updating RPC doc as well here? Informing that only child is also allowed.

https://github.com/bitcoin/bitcoin/blob/master/src/rpc/mempool.cpp#L926

Reply:
I had similar thoughts, yes. I think the current description is correct but less helpful than it could be (even without this PR).

Suggested alternative:

"The package must solely consist of a child transaction and all, some or none of its unconfirmed parents. None of the parents may depend on each other.\n"

Reply:
took a slightly different tack, I think this is more correct. You *must* include unconfirmed parents, if any exist.

Reply:
> You _must_ include unconfirmed parents, if any exist.

Thanks, you're right. I was confused by the `IsChildWithParents` documentation stating:

> ...not all parents need to be present...

But that is of course just an incomplete, context-free check. The `MemPoolAccept::AcceptPackage` documentation and implementation indeed require all unconfirmed parents to be included. Your [suggestion](https://github.com/bitcoin/bitcoin/compare/72872c7073e99da0828e788b25520dcecf721c1b..92ed1bd586f7048f75b38024521696ff04e5f69c) seems better.

Perhaps including the no-parents case in the RPC example would be helpful too:

<details>
<summary>git diff on 92ed1bd586</summary>

```diff
diff --git a/src/rpc/mempool.cpp b/src/rpc/mempool.cpp
index 09a9d87c2c..3074b3994a 100644
--- a/src/rpc/mempool.cpp
+++ b/src/rpc/mempool.cpp
@@ -965,8 +965,8 @@ static RPCHelpMan submitpackage()
             },
         },
         RPCExamples{
-            HelpExampleRpc("submitpackage", R"(["rawtx1", "rawtx2"])") +
-            HelpExampleCli("submitpackage", R"('["rawtx1", "rawtx2"]')")
+            HelpExampleRpc("submitpackage", R"(["raw-parent-tx-1", "raw-parent-tx-2", "raw-child-tx"])") +
+            HelpExampleCli("submitpackage", R"('["raw-tx-without-unconfirmed-parents"]')")
         },
         [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
         {

```
</details>

Reply:
can do when touched again

Reply:
done